tests/functional/container/test_container.py

```diff
@@ -699,28 +699,42 @@ def test_spare_with_one_broken(self):
         obj_meta = self.json_loads(resp.data)
         # Get a list of chunks for future spare request
         chunks = obj_meta['chunks']
-        if len(self.conf['services']['rawx']) < len(chunks) + 1:
+        if len(chunks) < count_broken + 1:
+            self.skip(
+                'Must run with a storage policy requiring more than %d chunk' %
+                count_broken)
+            return
+        elif len(self.conf['services']['rawx']) < len(chunks) + 1:
             self.skip('Not enough rawx services (%d+1 required)' % len(chunks))
             return
 
         # Extract one chunk from the list, keep it for later
-        broken = chunks.pop()
+        broken = [chunks.pop() for _ in range(count_broken)]
 
         # Do the spare request, specify that we already know some chunks,
         # and we know some chunk location is broken.
         resp = self.request('POST', self.url_content('spare'), params=params,
                             data=json.dumps(
                                 {"notin": chunks,
-                                 "broken": [broken]}))
+                                 "broken": broken}))
         self.assertEqual(resp.status, 200)
         spare_data = self.json_loads(resp.data)
         # Since we extracted one chunk, there must be exactly one chunk in
         # the response (plus one property telling the "quality" of the chunk)
-        self.assertEqual(1, len(spare_data['chunks']))
-        self.assertEqual(1, len(spare_data['properties']))
-        broken_netloc = broken['url'].split('/')[2]
-        spare_netloc = spare_data['chunks'][0]['id'].split('/')[2]
-        self.assertNotEqual(broken_netloc, spare_netloc)
+        self.assertEqual(count_broken, len(spare_data['chunks']))
+        self.assertEqual(count_broken, len(spare_data['properties']))
+        broken_netlocs = {b['url'].split('/')[2] for b in broken}
+        spare_netlocs = {s['id'].split('/')[2] for s in spare_data['chunks']}
+        self.assertNotEqual(broken_netlocs, spare_netlocs)
```

Comment:
This test is wrong: we have to check there are no common elements in both sets.
Do not merge before I fix this!